service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/MessageJudgeControllerAboutToStartTest.java

```diff
@@ -39,23 +40,44 @@ void shouldBuildC2DynamicListIfCaseDataHasC2Documents() {
                         .filename("c2_additional.doc")
                         .build())
                     .build())))
+            .judicialMessages(List.of(
+                element(JudicialMessage.builder()
+                    .latestMessage("some note")
+                    .messageHistory("some history")
+                    .dateSent("Some date sent")
+                    .build()),
+                element(JudicialMessage.builder()
+                    .latestMessage("some note")
+                    .messageHistory("some history")
+                    .dateSent("Some date sent")
+                    .build())))
             .build();
 
         AboutToStartOrSubmitCallbackResponse response = postAboutToStartEvent(asCaseDetails(caseData));
-        DynamicList builtDynamicList = mapper.convertValue(
+
+        DynamicList c2DynamicList = mapper.convertValue(
             response.getData().get("c2DynamicList"), DynamicList.class
         );
 
-        assertThat(builtDynamicList).isEqualTo(caseData.buildC2DocumentDynamicList());
+        DynamicList judicialMessageDynamicList = mapper.convertValue(
+            response.getData().get("judicialMessageDynamicList"), DynamicList.class
+        );
+
+        assertThat(c2DynamicList).isEqualTo(caseData.buildC2DocumentDynamicList());
+        assertThat(judicialMessageDynamicList).isEqualTo(caseData.buildJudicialMessageDynamicList());
```

Comment:
I don't like `caseData.buildJudicialMessageDynamicList()` for the assertion. What you are essentially doing is `        assertThat(caseData.buildJudicialMessageDynamicList()).isEqualTo(caseData.buildJudicialMessageDynamicList())`. You should instead manually construct the dynamic list in the test as you know what it should be.